frontend/components/profile/ProfilePicture.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState } from "react";
+import { StyleSheet, Text, View } from "react-native";
+import React from "react";
+import { TouchableOpacity } from "react-native";
+import { Image } from "react-native";
+import axios from "axios";
+
+import * as MediaLibrary from "expo-media-library";
+import * as ImagePicker from "expo-image-picker";
+
+import { RNS3 } from "react-native-aws3";
+import { useAuthContext } from "@/components/AuthProvider";
+type Props = {
+  uri: string;
+  editing: boolean;
+};
+
+const ProfilePicture = ({ uri, editing }: Props) => {
+  const { userId } = useAuthContext();
+  const [activeUri, setUri] = useState(uri);
+
+  const pick = async () => {
+    let result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.All,
+      allowsEditing: false,
+      allowsMultipleSelection: true,
+      quality: 1,
+      orderedSelection: true,
+    });
+    console.log(result);
+    if (!result.canceled) {
+      onDone(result.assets, true);
+    }
+  };
+
+  const onDone = (assets: any, isSelected: boolean) => {
+    console.log(assets);
+    console.log(isSelected);
+
+    const [asset] = assets;
+    const { uri, fileName, mimeType } = asset;
+
+    const file = {
+      uri: uri,
+      name: fileName,
+      type: mimeType,
+    };
+
+    const options = {
+      bucket: "playground-bucket-beak",
+      region: "us-east-2",
+      accessKey: process.env.EXPO_PUBLIC_AWS_ACCESS_KEY_ID as string,
+      secretKey: process.env.EXPO_PUBLIC_AWS_SECRET_ACCESS_KEY as string,
+      successActionStatus: 201,
+    };
+
+    RNS3.put(file, options)
```

Comment:
j curious - what made you decide to call s3 from the frontend and not the backend? noticed it's a lil slow to update the picture on the FE and am wondering if doing everything in one trip to the backend would be any speedier

Reply:
i think in a larger scale system throwing it in the backend might be more secure to do validation stuff- but i think otherwise you have to send the whole file to our backend and then again to s3 but not positive about the performance implications. doing it from the backend also requires handling aws's presigned url s3 ourselves which might make things more complicated also - but i agree having 2 calls (1 to aws, then to our backend) feels a little clunky

Reply:
makes sense! I think this works for now